backend_plugin/src/test/java/negatives/evaluators/TrailingWhiteSpaceNegative.java

```diff
@@ -0,0 +1,183 @@
+			e.printStackTrace();
+		}
+	}
+	
+	/**
+	 * Tests the user deleting the two trailing white spaces at the end
+	 * of the second line with two single backspaces
+	 */
+	@Test
+	public void removeAllTrailingSpacesTwoSpacesTwoBackspaces() {
+		try {
+			
+			// Start at the end of the second line
+			int offset = doc.getLineOffset(2) - 1;
+			
+			// Remove both white spaces (should trigger on the second)
+			assertFalse(mockUserSingleBackspace(offset--));
+			assertTrue(mockUserSingleBackspace(offset--));
+		} catch (BadLocationException e) {
+			// Should never get here
+			fail("Should never see this error in: " + this.getClass().getSimpleName() + "::" + this.getClass().getName());
+			e.printStackTrace();
+		}
+	}
+	
+	/**
+	 * Tests the user deleting the two trailing white spaces at the end
+	 * of the second line by selecting the two white spaces and hitting
+	 * backspace once
+	 */
+	@Test
+	public void removeAllTrailingSpacesTwoSpacesOneBackspace() {
+		try {
+			
+			// Start at the end of the second line
+			int offset = doc.getLineOffset(2) - 1;
+			
+			// Remove both white spaces with a single select/backspace combo
+			assertTrue(mockUserSelectingTwoCharactersAndBackspacing(offset));
+		} catch (BadLocationException e) {
+			// Should never get here
+			fail("Should never see this error in: " + this.getClass().getSimpleName() + "::" + this.getClass().getName());
+			e.printStackTrace();
+		}
+	}
+	
+	/**
+	 * Tests the user deleting the one trailing tab at the end of the third line
+	 */
+	@Test
+	public void removeAllTrailingTabsOneTab() {
+		try {
+			
+			// Start at the end of the third line
+			int offset = doc.getLineOffset(3) - 1;
+			
+			// Remove the single tab with a backspace
+			assertTrue(mockUserSingleBackspace(offset--));
+		} catch (BadLocationException e) {
+			// Should never get here
+			fail("Should never see this error in: " + this.getClass().getSimpleName() + "::" + this.getClass().getName());
+			e.printStackTrace();
+		}
+	}
+	
+	/**
+	 * Tests the user deleting the two trailing tabs at the end of the fourth line
+	 * with two single backspaces
+	 */
+	@Test
+	public void removeAllTrailingTabsTwoTabsTwoBackspaces() {
+		try {
+			
+			// Start at the end of the third line
+			int offset = doc.getLineOffset(4) - 1;
+			
+			// Remove the tabs with two backspaces (should trigger on the second)
+			assertFalse(mockUserSingleBackspace(offset--));
+			assertTrue(mockUserSingleBackspace(offset--));
+		} catch (BadLocationException e) {
+			// Should never get here
+			fail("Should never see this error in: " + this.getClass().getSimpleName() + "::" + this.getClass().getName());
+			e.printStackTrace();
+		}
+	}
+	
+	/**
+	 * Tests the user deleting the two trailing tabs at the end of the fourth line
+	 * by selecting the two tabs and hitting backspace once
+	 */
+	@Test
+	public void removeAllTrailingTabsTwoTabsOneBackspace() {
```

Comment:
Good job catching the case where multiple whitespace characters are deleted at the same time.